include/phasar/Utils/Utilities.h

```diff
@@ -75,6 +76,52 @@ struct stringIDLess {
   bool operator()(const std::string &lhs, const std::string &rhs) const;
 };
 
+/// Based on the reference implementation of std::remove_if
+/// "https://en.cppreference.com/w/cpp/algorithm/remove" and optimized for the
+/// case that a sorted list of indices is given instead of an unary predicate
+/// specifying the elements to be removed.
+template <typename It, typename EndIt, typename IdxIt, typename IdxEndIt>
+It remove_by_index(It First, EndIt Last, IdxIt FirstIndex, IdxEndIt LastIndex) {
+  if (FirstIndex == LastIndex || First == Last) {
+    return Last;
+  }
+  First = std::next(First, *FirstIndex);
+  if (First == Last) {
+    return First;
+  }
+
+  auto CurrIdx = *FirstIndex;
+
+  /// TODO: Optimize this loop
```

Comment:
Do you want to finish this before merging or is this a TODO for later?

Reply:
This is rather a TODO for later; I am just sure that we can improve this loop especially for trivially movable types, e.g. utilizing memcpy. However, this does not seem to be urgent